### PR TITLE
Add Segment.Size() method

### LINKED ISSUE #45: Cheap method to retrieve num documents in a Segment
Need a method on `Segment` which returns an estimate for the number of documents it contains. It can be a rough approximation. 

Intent is to use this during a background process in m3db (ticking) to emit metrics ~= cardinality.

### PR DESCRIPTION
**Pull request by prateek**
Resolves #45

### CODE REVIEW

File: index/segment/mem/segment.go [hunk 1]
```diff
@@ -82,6 +82,10 @@ func NewSegment(offset postings.ID, opts Options) (sgmt.MutableSegment, error) {
 	return s, nil
 }
 
+func (s *segment) Size() int64 {
+	return int64(int(s.readerID.Load()) - s.offset)
```
Comment (robskillington) [inline index/segment/mem/segment.go]: nit: Might be cleaner to `return s.readerID.Load() - int64(s.offset)` to protect against precision (obviously not going to practically be a problem, so take it or leave it).

Comment (prateek) [reply index/segment/mem/segment.go]: Sure thing.

Comment (prateek) [reply index/segment/mem/segment.go]: done

File: index/segment/mem/segment.go [hunk 2]
```diff
@@ -82,6 +82,10 @@ func NewSegment(offset postings.ID, opts Options) (sgmt.MutableSegment, error) {
 	return s, nil
 }
 
+func (s *segment) Size() int64 {
+	return int64(s.readerID.Load()) - int64(s.offset)
```
[the remark this answers is not in the document]
Comment (prateek) [reply index/segment/mem/segment.go]: talked offline: don't strictly need to, but better to do it for consistency across methods. addressed in https://github.com/m3db/m3ninx/pull/50

**Pull request merged**